### PR TITLE
Adds opinfo for pdist

### PR DESCRIPTION
**Pull request by mruberry**
Per title

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -16287,6 +16302,13 @@ def generate_std_var_kwargs(t: torch.Tensor, **kwargs):
             DecorateInfo(unittest.skip("Skipped!"), "TestJit", "test_variant_consistency_jit", dtypes=(torch.float32,),),
         )
     ),
+    OpInfo(
+        "nn.functional.pdist",
+        ref=reference_pdist,
+        sample_inputs_func=sample_inputs_pdist,
+        dtypes=floating_types(),
+        supports_autograd=False,
```
Comment (ngimel) [inline torch/testing/_internal/common_methods_invocations.py:16310]: `pdist` not supporting autograd is news to me

Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:16310]: I'll take a closer look

Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:16565]: Good catch

**Review by ngimel — APPROVED** [review]

**Pull request closed without merging**